sphere/initial_response.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * initial_response.php
+ *
+ * Special script to allow callback from Sphere to avoid cross origin breakage.
+ * Csrf security is maintained.
+ * Call to top.restoreSession() happens to ensure directed to correct session.
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Brady Miller <brady.g.miller@gmail.com>
+ * @copyright Copyright (c) 2021 Brady Miller <brady.g.miller@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+if (!empty($_GET['cancel']) && ($_GET['cancel'] == 'cancel')) {
```

Comment:
Does Sphere have any kind of signature validation for you to validate that the request originated from them?  We probably only want to process requests if they originated from sphere.